driver/src/main/java/org/neo4j/driver/v1/CypherType.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.v1;
+
+/**
+ * The Cypher type of a {@link Value}
+ */
+public interface CypherType
+{
+    /**
+     * @return the name of the Cypher type (as defined by Cypher)
+     */
+    String name();
+
+    /**
+     * @return the erased Cypher type of this type
+     */
+    CoarseCypherType erasedType();
```

Comment:
I'd love this JavaDoc to have a short description of what this means, because I don't think that's obvious to people who are not used to type theory.